src/main/java/com/pubnub/api/workers/SubscribeMessageProcessor.java

```diff
@@ -166,22 +168,10 @@ private BasePubSubResult getBasePubSubResult(SubscribeMessage message, String ch
     }
 
     private boolean isMessage(SubscribeMessage message) {
+        //should be "==" instead of "equals" for PNMessageType.MESSAGE01.getEValueFromServer()  because its value is null
         return message.getPnMessageType() == PNMessageType.MESSAGE01.getEValueFromServer() || message.getPnMessageType().equals(PNMessageType.MESSAGE02.getEValueFromServer());
```

Comment:
I know that it's not something that has changed in this commit but this comment with `==` triggered me. Why not return PNMessageType from the `getPnMessageType` already and make the check on enums instead?

Here's the changes that we might do in SubscribeMessage:

```

    private static Set<PNMessageType> PNMessageTypesSupportingEncryption = new HashSet<>(Arrays.asList(PNMessageType.MESSAGE01,
            PNMessageType.MESSAGE02,
            PNMessageType.FILE));

    public PNMessageType getPnMessageType() {
        return PNMessageType.valueByPnMessageType(pnMessageType);
    }

    public boolean supportsEncryption() {
        return PNMessageTypesSupportingEncryption.contains(getPnMessageType());
    }
```

and in here it would look like this (no need for comments):

```
    private boolean isMessage(SubscribeMessage message) {
        return message.getPnMessageType() == PNMessageType.MESSAGE01 || message.getPnMessageType() == PNMessageType.MESSAGE02;
    }
```

Reply:
Makes sense. I will modify it.